addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java

```diff
@@ -60,11 +61,14 @@ public synchronized void onConnectionOpened() {
     public synchronized void connect(String hostName, int port, ScheduledExecutorService scheduler) {
         try {
             wsUri = new URI(String.format("ws://%s:%d/jsonrpc", hostName, port));
+        } catch (URISyntaxException e) {
+            logger.error("exception during constructing URI host={}, port={}", hostName, port, e);
+        }
```

Comment:
imho you should not try to connect after this exception, maybe the catch block should be bigger, wdyt?